ARCHITECTURE.md

```diff
@@ -0,0 +1,275 @@
+# HyperDB Architecture
+
+HyperDB is a scalable peer-to-peer key-value database.
+
+## Filesystem metaphor
+
+HyperDB is structured to be used much like a traditional hierarchical
+filesystem. A value can be written and read at locations like `/foo/bar/baz`,
+and the API supports querying or tracking values at subpaths, like how watching
+for changes on `/foo/bar` will report both changes to `/foo/bar/baz` and also
+`/foo/bar/19`.
+
+## Set of append-only logs
+
+A HyperDB is fundamentally a set of
+[hypercore](https://github.com/mafintosh/hypercore)s. A *hypercore* is a secure
+append-only log that is identified by a public key, and can only be written to
+by the holder of the corresponding private key. Because it is append-only, old
+values cannot be deleted nor modified. Because it is secure, any other peer in
+the network can verify these attributes without inherently trusting the author.
+
+Each entry in a hypercore has a *sequence number*, that increments by 1 with
+each write, starting at 0 (`seq=0`).
+
+HyperDB builds its hierarchical key-value store on top of these hypercores, and
+also provides facilities for authorization, and replication of those member
+hypercores.
+
+### Directed acyclic graph
+
+The combination of all operations performed on a HyperDB by all of its members
+forms a DAG (*directed acyclic graph*). Each database operations link backward
```

Comment:
Each database *operation links* backwards...

*or*

*All* database operations link backward

Reply:
Took another swing at this. Read better?